sources/src/ChatGPTPopover.tsx

```diff
@@ -1,79 +1,143 @@
+        display: 'flex',
+        flexDirection: 'row',
+        placeContent: 'space-between',
+        alignItems: 'center',
+        pl: 2,
+        pr: 1,
+        py: 1,
+        ...appBarProps?.sx
+      }}
+    >
+      {children}
+    </AppBar>
+  );
+}
 
 export interface ChatGPTPopoverProps extends PopoverProps {
   appBarTitle?: string;
   width?: number | string;
   height?: number | string;
   appBarProps?: AppBarProps;
-  chatGPTProps: ChatGPTProps;
+  chatGPTProps?: ChatGPTProps;
+  isMinimized?: boolean;
+  onMinimize?: () => void;
+  onMaximize?: () => void;
 }
 
-function ChatGPTPopover(props: ChatGPTPopoverProps) {
+function ChatGPTPopover(props: Readonly<ChatGPTPopoverProps>) {
   const {
     open,
     onClose,
     appBarProps,
     chatGPTProps,
+    isMinimized = false,
+    onMinimize,
+    onMaximize,
     appBarTitle = 'AI Assistant',
     width = 450,
     height = 500,
     ...popoverProps
   } = props;
+
+  const handleMinimize = useCallback(() => {
+    onMinimize?.();
+  }, [onMinimize]);
+
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'Escape' && !isMinimized) {
+        handleMinimize();
+      }
+    };
+
+    if (open) {
+      document.addEventListener('keydown', handleKeyDown);
+    }
+
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [open, isMinimized, handleMinimize]);
+
   return (
-    <Popover
-      open={open}
-      onClose={onClose}
-      keepMounted={false}
-      anchorReference="none"
-      anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
-      anchorPosition={{ top: 100, left: 100 }}
-      BackdropProps={{ invisible: false, sx: { backgroundColor: 'rgba(0, 0, 0, 0.5)' } }}
-      {...popoverProps}
-      sx={{
-        [`> .${paperClasses.root}`]: {
-          width,
-          height,
-          display: 'flex',
-          flexDirection: 'column',
-          bottom: 10,
-          right: 10
-        },
-        ...popoverProps?.sx
-      }}
-    >
-      <AppBar
-        position="static"
-        {...appBarProps}
+    <>
+      <Popover
+        open={open}
+        onClose={onClose}
+        disableEscapeKeyDown={true}
+        keepMounted={false}
+        anchorReference="none"
+        anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
+        anchorPosition={{ top: 100, left: 100 }}
+        BackdropProps={{
+          invisible: false,
+          sx: { backgroundColor: 'rgba(0, 0, 0, 0.5)' },
+          onClick: (event) => {
+            event.stopPropagation();
+            handleMinimize();
+          }
+        }}
+        {...popoverProps}
         sx={{
-          display: 'flex',
-          flexDirection: 'row',
-          placeContent: 'space-between',
-          alignItems: 'center',
-          pl: 2,
-          pr: 1,
-          py: 1,
-          ...appBarProps?.sx
+          visibility: isMinimized ? 'hidden' : 'visible',
+          zIndex: 1400,
```

Comment:
What do you need to change the zIndex for? What are you trying to be on top of?
Should use `theme.zIndex.*` (e.g. `theme.zIndex.modal` or `theme.zIndex.modal + 1`)

Reply:
@rart I added this to avoid a situation where the chat popup appears behind the minimized bar. See the screenshot:
![Screenshot from 2024-10-29 16-20-49](https://github.com/user-attachments/assets/abd85b5f-d735-4288-b9b5-2506a61204e6)
`theme.zIndex.modal + 1` should work. How do we get the `theme` import from the plugin?
I tried with `import { useTheme } from '@mui/material/styles';` but this does not work as in the rollup, this translates to `craftercms.libs.MaterialUI.styles` where there is no export for this.

Reply:
You can...
- import { useTheme } from '@mui/material'
- or if you're within an `sx` block, you should be able to do 
  - `sx={(theme) => ({ ...theme.whatever... })}` or
  - `sx={{ something: (theme) => ({ ...theme.whatever... }) }}`

Reply:
I updated this. Thanks.